schema.py

```diff
@@ -126,21 +123,10 @@ def validate(self, data):
                     except SchemaError:
                         pass
                     else:
-                        try:
-                            nvalue = Schema(svalue, error=e).validate(value)
-                        except SchemaError as _x:
-                            x = _x
-                            raise
-                        else:
-                            coverage.add(skey)
-                            valid = True
-                            break
-                if valid:
-                    new[nkey] = nvalue
-                elif skey is not None:
-                    if x is not None:
-                        raise SchemaError(['Invalid value for key %r' % key] +
-                                          x.autos, [e] + x.errors)
```

Comment:
Concluding from my comment above, we can assert that this raise-statement is dead code.

Removing it cannot possibly break anything.

I have checked that the code affected by this change is essentially the same as in [0.3.1](https://github.com/keleshev/schema/blob/0.3.1/schema.py#L107-L138) which makes it unlikely that the change will uncover any problems in the wild.